src/transformers/cache_utils.py

```diff
@@ -1290,103 +1181,14 @@ def __init__(
         else:
             raise ValueError(f"Unknown quantization backend `{backend}`")
 
+        config = config.get_text_config()
```

Comment:
```suggestion
        config = config.get_text_config(decoder=True)
```
(related to https://github.com/huggingface/transformers/pull/40277)